src/Joystick/Joystick.cc

```diff
@@ -470,7 +477,7 @@ void Joystick::run()
         _update();
         _handleButtons();
         _handleAxis();
-        QGC::SLEEP::msleep(20);
+        QGC::SLEEP::msleep(qMin(static_cast<int>(1000.0f / _maxAxisFrequencyHz), static_cast<int>(1000.0f / _maxButtonFrequencyHz)) / 2);
```

Comment:
This duplicates the previous value of 20 at current defaults. But adjusts if you futz with the max Hz values.